pkg/cache/placeholder_manager.go

```diff
@@ -46,10 +47,13 @@ type PlaceholderManager struct {
 }
 
 var placeholderMgr *PlaceholderManager
+var placeholderManagerLock sync.Mutex
```

Comment:
The init and usage of the manager has not changed. Having a lock like this seems weird because we should only ever the one that was created by the current test, or when the context is created in production.
This probably also explains why you needed the nil check in the app code before.
This is a test problem. By adding the manager usage to the fail application test we are re-using the manager from a previous test. We should not be doing that. Please check the way `TestTryReserve()` and `TestTryReservePostRestart()` initiate the manager.
It looks like the test context does not create a manager and we might not clean up correctly either

Reply:
You are exactly correct. I have fixed the `TestFailApplication` following the example of `TestTryReserve`